airflow/providers/opsgenie/hooks/opsgenie.py

```diff
@@ -126,7 +126,7 @@ def close_alert(
                 identifier=identifier,
                 identifier_type=identifier_type,
                 close_alert_payload=close_alert_payload,
-                kwargs=kwargs,
+                **kwargs,
```

Comment:
Can you please also fix it in the operator https://github.com/apache/airflow/blob/22b49d334ef0008be7bd3d8481b55b8ab5d71c80/airflow/providers/opsgenie/operators/opsgenie.py#L209

I don't remember why I added it to begin with :/ 
Taking another look there are no additional parameters to pass to `close_alert` so this will only be useful if in the future the sdk will add more params.

Reply:
Right now, `OpsgenieCloseAlertOperator.execute` sends `close_alert_kwargs` as a normal param called `kwargs` ( which is not great, but should work ).

If we want to update that, this needs to be updated as well:

https://github.com/apache/airflow/blob/22b49d334ef0008be7bd3d8481b55b8ab5d71c80/airflow/providers/opsgenie/hooks/opsgenie.py#L106

So should I properly `kwarg`-ify the whole chain? including `OpsgenieAlertHook.close_alert`?

Reply:
ah right. I think it's fine because it's kwargs for `close_alert` so it's very self explanatory.
OK I remembered what I tried to do. the goal was to support `call_api` params
https://github.com/opsgenie/opsgenie-python-sdk/blob/0d20d2314522fc0fd8ca5f0faa16f7c96387e123/opsgenie_sdk/api_client.py#L380-L386
like `async_req`

It could be that the implementation is bugged. can you look into it?

Reply:
OK I took a deeper look
According to the source code the function accepts kwargs

https://github.com/opsgenie/opsgenie-python-sdk/blob/0d20d2314522fc0fd8ca5f0faa16f7c96387e123/opsgenie_sdk/api/alert/__init__.py#L913

and it does allow to set `async_req`

https://github.com/opsgenie/opsgenie-python-sdk/blob/0d20d2314522fc0fd8ca5f0faa16f7c96387e123/opsgenie_sdk/api/alert/__init__.py#L931

Am I missing something here?

Reply:
Hello guys,

Indeed we need to unpack and give extra params to the `AlertApi.close_alert`. The issue here is by not unpacking when calling the function, and giving directly `kwargs=kwargs`. (as a standard argument). It gets collected as `{kwargs: kwargs}`, which is not what we want.

Unpacking like this should fix the issue.

I think we should change how the parent function pass values to the kwargs parameter as it is a little bit confusing and unconventional.

```
# in OpsgenieCloseAlertOperator.execute
        self.hook.close_alert(
            identifier=self.identifier,
            identifier_type=self.identifier_type,
            payload=self._build_opsgenie_close_alert_payload(),
            **self.close_alert_kwargs,
        )
```

```
# In OpsgenieAlertHook.close_alert
    def close_alert(
        self,
        identifier: str,
        identifier_type: Optional[str] = 'id',
        payload: Optional[dict] = None,
        **kwargs,
    ) -> SuccessResponse:
```

Reply:
Thanks @pierrejeambrun !
@bperson would you apply the needed changes?